servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -795,6 +795,8 @@ void TextureStorage::texture_2d_layered_initialize(RID p_texture, const Vector<R
 		case RS::TEXTURE_LAYERED_CUBEMAP_ARRAY: {
 			texture.rd_type = RD::TEXTURE_TYPE_CUBE_ARRAY;
 		} break;
+		default:
+			ERR_FAIL(); // Shouldn't happen, silence warnings.
```

Comment:
The alternative is to initialize this variable before the switch even though it should be properly initialized in all switch branches.

The problem as I understand it is that enums suck :) So even if we make sure only those three values are passed as input, in theory something could call with a garbage int and it would work just fine, using uninitialized `texture.rd_type`.